File: web/features/school/domain/entities/school.ts
export type SchoolRole = "owner" | "teacher" | "admin";
export type SchoolType = "public" | "private" | "charter";

export type School = {
  id: string;
  name: string;
  logoImage: string | null;
  type: SchoolType;
  phone: string;
  role?: SchoolRole;
};

File: mobile/lib/features/main/domain/entities/school.dart
class School {
  final String id;
  final String name;
  final String? logoImage;
  final String type;
  final String phone;
  final String? role;

  School({
    required this.id,
    required this.name,
    required this.logoImage,
    required this.type,
    required this.phone,
    required this.role,
  });

  factory School.fromJson(Map<String, dynamic> json) => School(
    id: json["id"],
    name: json["name"],
    logoImage: json["logo_image"],
    type: json["type"],
    phone: json["phone"],
    role: json["role"],
  );

  Map<String, dynamic> toJson() => {
    "id": id,
    "name": name,
    "logo_image": logoImage,
    "type": type,
    "phone": phone,
    "role": role,
  };
}
